tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -25,29 +25,34 @@ LOG_MODULE_REGISTER(spi_loopback);
 #define MODE_LOOP 0
 #endif
 
-#define SPI_OP SPI_OP_MODE_MASTER | SPI_MODE_CPOL | MODE_LOOP | \
-	       SPI_MODE_CPHA | SPI_WORD_SET(8) | SPI_LINES_SINGLE
+#define SPI_OP(frame_size) SPI_OP_MODE_MASTER | SPI_MODE_CPOL | MODE_LOOP | \
+	       SPI_MODE_CPHA | SPI_WORD_SET(frame_size) | SPI_LINES_SINGLE
 
 
-static struct spi_dt_spec spi_fast = SPI_DT_SPEC_GET(SPI_FAST_DEV, SPI_OP, 0);
-static struct spi_dt_spec spi_slow = SPI_DT_SPEC_GET(SPI_SLOW_DEV, SPI_OP, 0);
+static struct spi_dt_spec spi_fast = SPI_DT_SPEC_GET(SPI_FAST_DEV, SPI_OP(8), 0);
+static struct spi_dt_spec spi_slow = SPI_DT_SPEC_GET(SPI_SLOW_DEV, SPI_OP(8), 0);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+static __used struct spi_dt_spec spi_fast_16 = SPI_DT_SPEC_GET(SPI_FAST_DEV, SPI_OP(16), 0);
+static __used struct spi_dt_spec spi_slow_16 = SPI_DT_SPEC_GET(SPI_SLOW_DEV, SPI_OP(16), 0);
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi) */
 
 /* to run this test, connect MOSI pin to the MISO of the SPI */
 
 #define STACK_SIZE 512
-#define BUF_SIZE 17
+#define BUF_SIZE 18
```

Comment:
I need to change this to 18 because 17 is not divisible by 2, therefore it can't be sent with 16 bits frames if I'm not mistaken.